web/src/actions/state.js

```diff
@@ -0,0 +1,71 @@
+import axios from '../util/api';
+
+export const GET_STATE_REQUEST = 'GET_STATE_REQUEST';
+export const GET_STATE_SUCCESS = 'GET_STATE_SUCCESS';
+export const GET_STATE_FAILURE = 'GET_STATE_FAILURE';
+
+export const UPDATE_STATE_REQUEST = 'UPDATE_STATE_REQUEST';
+export const UPDATE_STATE_SUCCESS = 'UPDATE_STATE_SUCCESS';
+export const UPDATE_STATE_FAILURE = 'UPDATE_STATE_FAILURE';
+
+export const requestState = () => ({ type: GET_STATE_REQUEST });
+export const receiveState = data => ({ type: GET_STATE_SUCCESS, data });
+export const failState = error => ({ type: GET_STATE_FAILURE, error });
+
+export const requestStateUpdate = () => ({ type: UPDATE_STATE_REQUEST });
+export const receiveStateUpdate = data => ({
+  type: UPDATE_STATE_SUCCESS,
+  data
+});
+export const failStateUpdate = error => ({ type: UPDATE_STATE_FAILURE, error });
+
+export const fetchState = id => dispatch => {
+  dispatch(requestState());
+
+  const url = `/states${id ? `/${id}` : ''}`;
+
+  return axios
+    .get(url)
+    .then(req => dispatch(receiveState(req.data)))
+    .catch(error => {
+      const reason = error.response.data || 'N/A';
+      dispatch(failState(reason));
+    });
+};
+
+export const updateState = (id, data) => dispatch => {
+  dispatch(requestStateUpdate());
+
+  const url = `/states/${id}`;
+
+  if (data.medicaid_office) {
+    const { medicaid_office: { director } } = data;
```

Comment:
for these nested ones, i think `const { director } = data.medicaid_office` is a little easier to read

Reply:
feel free to agree to disagree